[core] Clarify release version bump strategy (#15219)
mui/mui-x: scripts/README.md

2. Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
3. Update the root `package.json`'s version
4. Update the versions of the other `package.json` files and of the dependencies with `pnpm release:version` (`pnpm release:version prerelease` for alpha / beta releases).

>    Do not skip the version bump if Lerna detects a change in the package. It is important to release the package if there are any** changes to it.

Reviewer: ```suggestion
   Do not skip the version bump if Lerna detects a change in the package. It is important to release the package if there are **any** changes to it.
```

```suggestion
   Do not skip the version bump if Lerna detects a change in the package. It is important to release the package if there are any changes to it.
```

I assume you wanted to highlight this word

>    Make sure to not skip the version bump if Lerna detected a change in the package. It is important to release the package if there were **any** changes to it.

Reviewer 2: ```suggestion
   Do not skip the version bump if Lerna detects a change in the package. It is important to release the package if there are any** changes to it.
```

>    If Lerna doesn't suggest a version bump for the package, there is no need to release it.

Reviewer 3: ```suggestion
   If Lerna doesn't suggest a version bump for the package, don't release it.
```

Reviewer 2: Indeed, that's even better. 👍

>    If Lerna doesn't suggest a version bump for the package, we don't release it.

Reviewer 2: Nitpick to avoid Vale error, but the existing wording works great as well. 👍 
```suggestion
   If Lerna doesn't suggest a version bump for the package, there is no need to release it.
```
